lib/backy/cli.rb

```diff
@@ -0,0 +1,75 @@
+require "thor"
+
+module Backy
+  class CLI < Thor
+    desc "backup", "Perform a backup operation using the current configuration settings"
+    def backup
+      result = Backy::PgDump.new.call
+      puts "Backup created: #{result}"
+    end
+
+    desc "push", "Push the local dump to a remote storage location"
+    def push
+      file_name = Backy::List.new.call.select(&:local?).last
+      Backy::S3Save.new(file_name: file_name).call
+      File.delete(file_name) if File.exist?(file_name)
+      puts "Backup pushed to remote storage: #{file_name}"
+    end
+
+    desc "upload FILE_NAME", "Upload a specified backup file to the remote storage"
+    def upload(file_name)
+      if file_name.nil? || !File.exist?(file_name)
+        puts "Please set DUMP_FILE env variable to the existing local file to save to S3"
+        exit
+      end
+
+      Backy::S3Save.new(file_name: file_name).call
+      puts "Backup uploaded to remote storage: #{file_name}"
+    end
+
+    desc "download FILE_NAME", "Download a specified backup file from the remote storage to the local system"
+    def download(file_name)
+      if file_name.nil? || File.exist?(file_name)
+        puts "Please set DUMP_FILE env variable to the missing remote file to download"
+        exit
+      end
+
+      Backy::S3Load.new(file_name: file_name).call
+      puts "Backup downloaded from remote storage: #{file_name}"
+    end
+
+    desc "list", "Display a list of available backups, both locally and in remote storage"
+    def list
+      list = Backy::List.call
```

Comment:
```suggestion
      list = Backy::List.new.call
```

I guess we could re introduce a `call` class method for convenience. I had to remove the base class because it was using new syntax not supported by older rubies.

Reply:
@str1fe Lets leave it at .new.call, its easier.